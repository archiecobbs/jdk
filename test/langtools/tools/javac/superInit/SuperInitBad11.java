/*
 * @test /nodynamiccopyright/
 * @bug 8194743
 * @summary Test invalid reference to "this" prior to superclass construction
 *
 * @compile/fail/ref=SuperInitBad11.out -XDrawDiagnostics SuperInitBad11.java
 */
class SuperInitBad11 implements Iterable<Byte> {

    SuperInitBad11() {
        Iterable.super.spliterator();
        super();
    }

    @Override
    public java.util.Iterator<Byte> iterator() {
        return null;
    }
}
