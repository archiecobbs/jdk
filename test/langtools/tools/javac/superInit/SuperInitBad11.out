SuperInitBad11.java:11:17: compiler.err.cant.ref.before.ctor.called: super
1 error
